website/docs/r/s3_bucket_policy.html.markdown

```diff
@@ -15,36 +15,33 @@ Attaches a policy to an S3 bucket resource.
 ### Basic Usage
 
 ```terraform
-resource "aws_s3_bucket" "b" {
+locals {
+  aws_account_b_id = "123456789012"
+}
+
+resource "aws_s3_bucket" "example" {
   bucket = "my-tf-test-bucket"
 }
 
-resource "aws_s3_bucket_policy" "b" {
-  bucket = aws_s3_bucket.b.id
-
-  # Terraform's "jsonencode" function converts a
-  # Terraform expression's result to valid JSON syntax.
-  policy = jsonencode({
-    Version = "2012-10-17"
-    Id      = "MYBUCKETPOLICY"
-    Statement = [
-      {
-        Sid       = "IPAllow"
-        Effect    = "Deny"
-        Principal = "*"
-        Action    = "s3:*"
-        Resource = [
-          aws_s3_bucket.b.arn,
-          "${aws_s3_bucket.b.arn}/*",
-        ]
-        Condition = {
-          NotIpAddress = {
-            "aws:SourceIp" = "8.8.8.8/32"
-          }
-        }
-      },
+resource "aws_s3_bucket_policy" "allow_access_from_account_b" {
+  bucket = aws_s3_bucket.example.id
+  policy = data.aws_iam_policy_document.allow_access_from_account_b.json
+}
+
+data "aws_iam_policy_document" "allow_access_from_account_b" {
+  statement {
+    principals {
+      type         = "AWS"
+      identitfiers = [local.aws_account_b_id]
```

Comment:
As noted above, I think adding a `locals` block might add more confusion for someone less familiar with Terraform, so it probably makes more sense to just pass this value directly.

```suggestion
      identitfiers = ["123456789012"]
```